app/utils/parseCacheControl.test.ts

```diff
@@ -0,0 +1,223 @@
+        'CDN-Cache-Control': 's-maxage=2000',
+        'Netlify-CDN-Cache-Control': 'max-age=3000',
+        'Age': '100',
+      })
+      const now = Date.now()
+
+      const result = parseCacheControl(headers, now)
+
+      expect(result.netlifyCdnTtl).toBe(2900) // 3000 - 100
+    })
+  })
+
+  describe('revalidate determination', () => {
+    it('returns must-revalidate when must-revalidate directive is present', () => {
+      const headers = createMockHeaders({
+        'Cache-Control': 'max-age=3600, must-revalidate',
+      })
+      const now = Date.now()
+
+      const result = parseCacheControl(headers, now)
+
+      expect(result.revalidate).toBe('must-revalidate')
+    })
+
+    it('returns immutable when immutable directive is present', () => {
+      const headers = createMockHeaders({
+        'Cache-Control': 'max-age=3600, immutable',
+      })
+      const now = Date.now()
+
+      const result = parseCacheControl(headers, now)
+
+      expect(result.revalidate).toBe('immutable')
+    })
+
+    it('prioritizes must-revalidate over immutable when both are present', () => {
+      const headers = createMockHeaders({
+        'Cache-Control': 'max-age=3600, must-revalidate, immutable',
+      })
+      const now = Date.now()
+
+      const result = parseCacheControl(headers, now)
+
+      expect(result.revalidate).toBe('must-revalidate')
+    })
+
+    it('returns undefined when neither must-revalidate nor immutable is present', () => {
+      const headers = createMockHeaders({
+        'Cache-Control': 'max-age=3600',
+      })
+      const now = Date.now()
+
+      const result = parseCacheControl(headers, now)
+
+      expect(result.revalidate).toBeUndefined()
+    })
+  })
+
+  describe('header parsing', () => {
+    it('parses Age header correctly', () => {
+      const headers = createMockHeaders({
+        Age: '1800',
+      })
+      const now = Date.now()
+
+      const result = parseCacheControl(headers, now)
+
+      expect(result.age).toBe(1800)
+    })
+
+    it('ignores empty Age header', () => {
+      const headers = createMockHeaders({
+        Age: '',
+      })
+      const now = Date.now()
+
+      const result = parseCacheControl(headers, now)
+
+      expect(result.age).toBeUndefined()
+    })
+
+    it('parses Date header correctly', () => {
+      const headers = createMockHeaders({
+        Date: 'Wed, 21 Oct 2015 07:28:00 GMT',
+      })
+      const now = Date.now()
+
+      const result = parseCacheControl(headers, now)
+
+      expect(result.date).toEqual(new Date('Wed, 21 Oct 2015 07:28:00 GMT'))
+    })
+
+    it('parses Expires header correctly', () => {
+      const headers = createMockHeaders({
+        Expires: 'Wed, 21 Oct 2015 08:28:00 GMT',
+      })
+      const now = Date.now()
+
+      const result = parseCacheControl(headers, now)
+
+      expect(result.expiresAt).toEqual(new Date('Wed, 21 Oct 2015 08:28:00 GMT'))
+    })
+
+    it('extracts ETag header correctly', () => {
```

Comment:
NEVER USE "correctly" or "expected". describe the behaviour. this applies to all tests.